File: app/src/test/java/com/metahumanz/pacilread/export/BookExportNamingTest.java
package com.metahumanz.pacilread.export;

import static org.junit.Assert.assertEquals;

import com.metahumanz.pacilread.model.BookRecord;

import org.junit.Test;

import java.util.HashSet;
import java.util.Set;

public class BookExportNamingTest {
    @Test
    public void preservesOriginalNameAndAddsCollisionSuffix() {
        BookRecord book = new BookRecord();
        book.sourceDisplayName = "小说.epub";
        book.bookType = "epub";
        book.localPath = "C:/books/random.epub";
        Set<String> used = new HashSet<>();
        used.add("小说.epub");

        assertEquals("小说 (2).epub", BookExportNaming.uniqueFileName(book, used));
    }

    @Test
    public void buildsSafeFallbackForLegacyBook() {
        BookRecord book = new BookRecord();
        book.title = "标题:一";
        book.author = "作者";
        book.bookType = "pdf";

        assertEquals("标题_一 - 作者.pdf", BookExportNaming.uniqueFileName(book, new HashSet<>()));
    }
}

File: app/src/test/kotlin/com/metahumanz/pacilread/export/BookExportNamingTest.kt
package com.metahumanz.pacilread.export

import com.metahumanz.pacilread.model.BookRecord
import org.junit.Assert.assertEquals
import org.junit.Test

class BookExportNamingTest {
    @Test
    fun preservesOriginalNameAndAddsCollisionSuffix() {
        val book = BookRecord().apply {
            sourceDisplayName = "小说.epub"
            bookType = "epub"
            localPath = "C:/books/random.epub"
        }
        assertEquals("小说 (2).epub", BookExportNaming.uniqueFileName(book, hashSetOf("小说.epub")))
    }

    @Test
    fun buildsSafeFallbackForLegacyBook() {
        val book = BookRecord().apply {
            title = "标题:一"
            author = "作者"
            bookType = "pdf"
        }
        assertEquals("标题_一 - 作者.pdf", BookExportNaming.uniqueFileName(book, hashSetOf()))
    }
}
